code/modules/hydroponics/hydroitemdefines.dm

```diff
@@ -140,13 +142,13 @@
  *
  * Returns the formatted output as text.
  */
-/obj/item/plant_analyzer/proc/scan_plant(obj/item/scanned_object)
+/obj/item/plant_analyzer/proc/scan_plant(obj/item/scanned_object, mode)
 	var/returned_message = "<span class='info'>*---------*\nThis is \a <span class='name'>[scanned_object]</span>.\n"
 	var/obj/item/seeds/our_seed = scanned_object
 	if(!istype(our_seed)) //if we weren't passed a seed, we were passed a plant with a seed
 		our_seed = scanned_object.get_plant_seed()
 
-	switch(scan_mode)
+	switch(mode)
```

Comment:
I'd much rather these be separate procs. There's no reason to share this behavior in this one area.